File: proj10/ruby/penguin.rb
# penguin.rb defines the behavior of a penguin.
# Author: David Reidsma
# Date: 04/30/2020

require './walking_bird.rb'

class Penguin < WalkingBird
  # Get the call of this bird.
  def call
    return "Blraww!"
  end
end

File: proj10/java/Penguin.java
/**
 * Penguin.java defines the behavior of a penguin.
 *
 * Completed by: David Reidsma
 * Date: 04/30/2020
 */

public class Penguin extends WalkingBird {
    /**
     * implicit constructor
     * Postcondition: myName = ""
     */
    public Penguin() {
	super();
    }

    /**
     * explicit constructor
     * @param name a String 
     * PostCond: myName == name.
     */
    public Penguin(String name) {
	super(name);
    }

    /**
     * Get this Bird's call.
     * @return The call of the bird.
     */
    public String getCall() {
	return "Blraww!";
    }
}
